src/sim-election.lisp

```diff
@@ -0,0 +1,138 @@
+;; election.lisp -- Leader Elections
+;;
+;; DM/Emotiq  03/18
+;; ---------------------------------------------------------------
+#|
+The MIT License
+
+Copyright (c) 2018 Emotiq AG
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+|#
+
+(in-package :sim-trial-election)
+;; ---------------------------------------------------------------
+
+(defvar *beacon-timer*     nil)
+;(defvar *beacon-interval*  30)
+(defvar *beacon-interval*  7)
+(defvar *self-destruct*    nil)
+
+(defun kill-beacon ()
+  (setf *self-destruct* t))
+
+;; mp is the multi-processing library for LispWorks
+(defun make-trial-election-beacon (handler-actor)
+  (unless *beacon-timer*
+    (setf *beacon-timer* (mp:make-timer (lambda ()
```

Comment:
@guitarvydas We have to make the code work under `ccl` (and hopefully soon `sbcl`).  I think you should use the `mp-compat` system here to make a timer, or go find a suitable system in Quicklisp